commit a89bd259ff20f15e9947f993157a16e4b087d5fe
Author: Florent Jardin <florent@jardin.im>
Date:   Sat Aug 8 13:05:40 2026 +0200

    printer: python3 drivers

diff --git a/xmlescpos/printer.py b/xmlescpos/printer.py
index 0847c82..66fc0f8 100644
--- a/xmlescpos/printer.py
+++ b/xmlescpos/printer.py
@@ -32,23 +32,25 @@ class Usb(Escpos):
         """ Search device on USB tree and set is as escpos device """
         self.device = usb.core.find(idVendor=self.idVendor, idProduct=self.idProduct)
         if self.device is None:
-            print "Cable isn't plugged in"
+            print("Cable isn't plugged in")
 
         if self.device.is_kernel_driver_active(0):
             try:
                 self.device.detach_kernel_driver(0)
             except usb.core.USBError as e:
-                print "Could not detatch kernel driver: %s" % str(e)
+                print("Could not detatch kernel driver: %s" % str(e))
 
         try:
             self.device.set_configuration()
             self.device.reset()
         except usb.core.USBError as e:
-            print "Could not set configuration: %s" % str(e)
+            print("Could not set configuration: %s" % str(e))
 
 
     def _raw(self, msg):
         """ Print any command sent in raw format """
+        if isinstance(msg, str):
+            msg = msg.encode('latin-1')
         self.device.write(self.out_ep, msg, self.interface)
 
 
@@ -79,16 +81,18 @@ class Serial(Escpos):
 
     def open(self):
         """ Setup serial port and set is as escpos device """
-        self.device = serial.Serial(port=self.devfile, baudrate=self.baudrate, bytesize=self.bytesize, parity=serial.PARITY_NONE, stopbits=serial.STOPBITS_ONE, timeout=self.timeout, dsrdtr=True)
+        self.device = serial.Serial(port=self.devfile, baudrate=self.baudrate, bytesize=self.bytesize, parity=serial.PARITY_NONE, stopbits=serial.STOPBITS_ONE, timeout=self.timeout)
 
         if self.device is not None:
-            print "Serial printer enabled"
+            print("Serial printer enabled")
         else:
-            print "Unable to open serial printer on: %s" % self.devfile
+            print("Unable to open serial printer on: %s" % self.devfile)
 
 
     def _raw(self, msg):
         """ Print any command sent in raw format """
+        if isinstance(msg, str):
+            msg = msg.encode('latin-1')
         self.device.write(msg)
 
 
@@ -118,10 +122,12 @@ class Network(Escpos):
         self.device.connect((self.host, self.port))
 
         if self.device is None:
-            print "Could not open socket for %s" % self.host
+            print("Could not open socket for %s" % self.host)
 
 
     def _raw(self, msg):
+        if isinstance(msg, str):
+            msg = msg.encode('latin-1')
         self.device.send(msg)
 
 
