commit fb89171961bc69661cfdc421778d503f9bfa10d1
Author: Florent Jardin <florent@jardin.im>
Date:   Sat Aug 8 13:04:51 2026 +0200

    escpos: python3 language fixes

diff --git a/xmlescpos/escpos.py b/xmlescpos/escpos.py
index f064cfa..0447059 100644
--- a/xmlescpos/escpos.py
+++ b/xmlescpos/escpos.py
@@ -28,8 +28,10 @@ from exceptions import *
 
 def utfstr(stuff):
     """ converts stuff to string and does without failing if stuff is a utf8 string """
-    if isinstance(stuff,basestring):
+    if isinstance(stuff, str):
         return stuff
+    elif isinstance(stuff, bytes):
+        return stuff.decode('utf-8', 'replace')
     else:
         return str(stuff)
 
@@ -135,7 +137,7 @@ class StyleStack:
         _style = {}
         for attr in style:
             if attr in self.cmds and not style[attr] in self.cmds[attr]:
-                print 'WARNING: ESC/POS PRINTING: ignoring invalid value: '+utfstr(style[attr])+' for style: '+utfstr(attr)
+                print('WARNING: ESC/POS PRINTING: ignoring invalid value: '+utfstr(style[attr])+' for style: '+utfstr(attr))
             else:
                 _style[attr] = self.enforce_type(attr, style[attr])
         self.stack.append(_style)
@@ -145,7 +147,7 @@ class StyleStack:
         _style = {}
         for attr in style:
             if attr in self.cmds and not style[attr] in self.cmds[attr]:
-                print 'WARNING: ESC/POS PRINTING: ignoring invalid value: '+utfstr(style[attr])+' for style: '+utfstr(attr)
+                print('WARNING: ESC/POS PRINTING: ignoring invalid value: '+utfstr(style[attr])+' for style: '+utfstr(attr))
             else:
                 self.stack[-1][attr] = self.enforce_type(attr, style[attr])
 
@@ -156,7 +158,7 @@ class StyleStack:
 
     def to_escpos(self):
         """ converts the current style to an escpos command string """
-        cmd = ''
+        cmd = b''
         for style in self.cmds:
             cmd += self.cmds[style][self.get(style)]
         return cmd
@@ -208,7 +210,7 @@ class XmlSerializer:
         if text:
             text = utfstr(text)
             text = text.strip()
-            text = re.sub('\s+',' ',text)
+            text = re.sub(r'\s+',' ',text)
             if text:
                 self.dirty = True
                 self.escpos.text(text)
@@ -273,7 +275,7 @@ class XmlLineSerializer:
         if text:
             text = utfstr(text)
             text = text.strip()
-            text = re.sub('\s+',' ',text)
+            text = re.sub(r'\s+',' ',text)
             if text:
                 self._txt(text)
 
@@ -371,7 +373,7 @@ class Escpos:
 
 
         if im.size[0] > 512:
-            print  "WARNING: Image is wider than 512 and could be truncated at print time "
+            print("WARNING: Image is wider than 512 and could be truncated at print time")
         if im.size[1] > 255:
             raise ImageSizeError()
 
@@ -520,7 +522,7 @@ class Escpos:
             if not string:
                 string = ''
             string = string.strip()
-            string = re.sub('\s+',' ',string)
+            string = re.sub(r'\s+',' ',string)
             return string
 
         def format_value(value, decimals=3, width=0, decimals_separator='.', thousands_separator=',', autoint=False, symbol='', position='after'):
