commit fb89171961bc69661cfdc421778d503f9bfa10d1
Author: Florent Jardin <florent@jardin.im>
Date:   Sat Aug 8 13:04:51 2026 +0200

    escpos: python3 language fixes

diff --git a/xmlescpos/escpos.py b/xmlescpos/escpos.py
index f064cfa..0447059 100644
--- a/xmlescpos/escpos.py
+++ b/xmlescpos/escpos.py
@@ -28,8 +28,10 @@ from exceptions import *
 
 def utfstr(stuff):
     """ converts stuff to string and does without failing if stuff is a utf8 string """
-    if isinstance(stuff,basestring):
+    if isinstance(stuff, str):
         return stuff
+    elif isinstance(stuff, bytes):
+        return stuff.decode('utf-8', 'replace')
     else:
         return str(stuff)
 
@@ -135,7 +137,7 @@ class StyleStack:
         _style = {}
         for attr in style:
             if attr in self.cmds and not style[attr] in self.cmds[attr]:
-                print 'WARNING: ESC/POS PRINTING: ignoring invalid value: '+utfstr(style[attr])+' for style: '+utfstr(attr)
+                print('WARNING: ESC/POS PRINTING: ignoring invalid value: '+utfstr(style[attr])+' for style: '+utfstr(attr))
             else:
                 _style[attr] = self.enforce_type(attr, style[attr])
         self.stack.append(_style)
@@ -145,7 +147,7 @@ class StyleStack:
         _style = {}
         for attr in style:
             if attr in self.cmds and not style[attr] in self.cmds[attr]:
-                print 'WARNING: ESC/POS PRINTING: ignoring invalid value: '+utfstr(style[attr])+' for style: '+utfstr(attr)
+                print('WARNING: ESC/POS PRINTING: ignoring invalid value: '+utfstr(style[attr])+' for style: '+utfstr(attr))
             else:
                 self.stack[-1][attr] = self.enforce_type(attr, style[attr])
 
@@ -156,7 +158,7 @@ class StyleStack:
 
     def to_escpos(self):
         """ converts the current style to an escpos command string """
-        cmd = ''
+        cmd = b''
         for style in self.cmds:
             cmd += self.cmds[style][self.get(style)]
         return cmd
@@ -208,7 +210,7 @@ class XmlSerializer:
         if text:
             text = utfstr(text)
             text = text.strip()
-            text = re.sub('\s+',' ',text)
+            text = re.sub(r'\s+',' ',text)
             if text:
                 self.dirty = True
                 self.escpos.text(text)
@@ -273,7 +275,7 @@ class XmlLineSerializer:
         if text:
             text = utfstr(text)
             text = text.strip()
-            text = re.sub('\s+',' ',text)
+            text = re.sub(r'\s+',' ',text)
             if text:
                 self._txt(text)
 
@@ -371,7 +373,7 @@ class Escpos:
 
 
         if im.size[0] > 512:
-            print  "WARNING: Image is wider than 512 and could be truncated at print time "
+            print("WARNING: Image is wider than 512 and could be truncated at print time")
         if im.size[1] > 255:
             raise ImageSizeError()
 
@@ -520,7 +522,7 @@ class Escpos:
             if not string:
                 string = ''
             string = string.strip()
-            string = re.sub('\s+',' ',string)
+            string = re.sub(r'\s+',' ',string)
             return string
 
         def format_value(value, decimals=3, width=0, decimals_separator='.', thousands_separator=',', autoint=False, symbol='', position='after'):
diff --git a/xmlescpos/printer.py b/xmlescpos/printer.py
index 0847c82..66fc0f8 100644
--- a/xmlescpos/printer.py
+++ b/xmlescpos/printer.py
@@ -32,23 +32,25 @@ class Usb(Escpos):
         """ Search device on USB tree and set is as escpos device """
         self.device = usb.core.find(idVendor=self.idVendor, idProduct=self.idProduct)
         if self.device is None:
-            print "Cable isn't plugged in"
+            print("Cable isn't plugged in")

         if self.device.is_kernel_driver_active(0):
             try:
                 self.device.detach_kernel_driver(0)
             except usb.core.USBError as e:
-                print "Could not detatch kernel driver: %s" % str(e)
+                print("Could not detatch kernel driver: %s" % str(e))

         try:
             self.device.set_configuration()
             self.device.reset()
         except usb.core.USBError as e:
-            print "Could not set configuration: %s" % str(e)
+            print("Could not set configuration: %s" % str(e))


     def _raw(self, msg):
         """ Print any command sent in raw format """
+        if isinstance(msg, str):
+            msg = msg.encode('latin-1')
         self.device.write(self.out_ep, msg, self.interface)


@@ -79,16 +81,18 @@ class Serial(Escpos):

     def open(self):
         """ Setup serial port and set is as escpos device """
-        self.device = serial.Serial(port=self.devfile, baudrate=self.baudrate, bytesize=self.bytesize, parity=serial.PARITY_NONE, stopbits=serial.STOPBITS_ONE, timeout=self.timeout, dsrdtr=True)
+        self.device = serial.Serial(port=self.devfile, baudrate=self.baudrate, bytesize=self.bytesize, parity=serial.PARITY_NONE, stopbits=serial.STOPBITS_ONE, timeout=self.timeout)

         if self.device is not None:
-            print "Serial printer enabled"
+            print("Serial printer enabled")
         else:
-            print "Unable to open serial printer on: %s" % self.devfile
+            print("Unable to open serial printer on: %s" % self.devfile)


     def _raw(self, msg):
         """ Print any command sent in raw format """
+        if isinstance(msg, str):
+            msg = msg.encode('latin-1')
         self.device.write(msg)


@@ -118,10 +122,12 @@ class Network(Escpos):
         self.device.connect((self.host, self.port))

         if self.device is None:
-            print "Could not open socket for %s" % self.host
+            print("Could not open socket for %s" % self.host)


     def _raw(self, msg):
+        if isinstance(msg, str):
+            msg = msg.encode('latin-1')
         self.device.send(msg)
